geopyspark/geotrellis/rasterio.py

```diff
@@ -0,0 +1,91 @@
+import os
+import math
+import geopyspark as gps
+try:
+    import rasterio
+except ImportError:
+    raise ImportError("rasterio must be installed in order to use the features in the geopyspark.geotrellis.rasterio package")
+
+
+__all__ = ['read_windows', 'get']
+
+# On driver
+_GDAL_DATA = os.environ.get("GDAL_DATA")
+
+def crs_to_proj4(crs):
+    """Converts RasterIO CRS to proj4 string using osgeo library.
+
+    Args:
+        crs: :class:`rasterio.crs.CRS`
+
+    Returns:
+        Proj4 string of CRS.
+    """
+    try:
+        from osgeo import osr
+    except ImportError:
+        raise ImportError("osgeo must be installed in order to use the crs_to_proj4 function")
+
+    srs = osr.SpatialReference()
+    srs.ImportFromWkt(crs.wkt)
+    proj4 = srs.ExportToProj4()
+    return proj4
+
+def read_windows(uri, xcols=512, ycols=512, bands=[1], crs_to_proj4=crs_to_proj4):
+    """Given a URI, this method uses rasterio to generate series of windows of the desired dimensions.
+
+    Args:
+        uri: The URI where the source data can be found.
+        xcols: The desired tile width.
+        ycols: The desired tile height.
+        bands: The bands from which windows should be produced.  An array of integers.
+        crs_to_proj4: A that takes a :class:`rasterio.crs.CRS` and returns a Proj4 string.
+
+    Returns:
+        Generator of (:class:`~geopyspark.geotrellis.ProjectedExtent`, :class:`~geopyspark.geotrellis.Tile`)
+    """
+
+    if ("GDAL_DATA" not in os.environ) and (_GDAL_DATA != None):
```

Comment:
I'm still not sure how `_GDAL_DATA` is set when it's not in the environment.

Reply:
Not sure the mechanics here, this is something I transcribed from James work.
This is intended to handle the case where `GDAL_DATA` is set on master but not the workers.
Presumably the only way it works if `_GDAL_DATA` is captured as a closures in `PipelinedRDD`. I'm honestly not sure how or if that process actually happens here.

Reply:
I see. I'm fine with keeping it in for now just to make sure everything works. We can look into removing it later.